src/Uno.Extensions.Toolkit.UI/WindowExtensions.cs

```diff
@@ -4,4 +4,26 @@ public static class WindowExtensions
 {
 	public static IThemeService GetThemeService(this Window window, ILogger? logger = default) =>
 		new ThemeService(() => window.Content?.XamlRoot, new Dispatcher(window), logger);
+
+	public static Task<IThemeService> GetInitializedThemeServiceAsync(this Window window, ILogger? logger = default) =>
+		window.InitializeWindowThemeServiceAsync(window.GetThemeService(logger));
+
+	internal static async Task<IThemeService> InitializeWindowThemeServiceAsync(this Window window, IThemeService themeService)
+	{
+		if (window.Content is null)
+		{
+			window.Activated += async (_,_)=>
+			{
+				if(window.Content is { } element)
+				{
+					await element.InitializeElementThemeServiceAsync(themeService);
+				}
+			};
```

Comment:
Same as with the FrameworkElement. Let's be responsible and unhook from the Activated event.